src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -473,6 +456,62 @@ public Long getAllowedTokenExpirationClockSkewSeconds() {
         return allowedTokenExpirationClockSkewSeconds;
     }
 
+    @PostConstruct
+    @Restricted(NoExternalUse.class)
+    public void createProxyAwareResourceRetriver() {
+        proxyAwareResourceRetriever =
+                ProxyAwareResourceRetriever.createProxyAwareResourceRetriver(isDisableSslVerification());
+    }
+
+    ProxyAwareResourceRetriever getResourceRetriever() {
+        return proxyAwareResourceRetriever;
+    }
+
+    private OidcConfiguration buildOidcConfiguration() {
+        // TODO cache this and use the well known if available.
+        OidcConfiguration conf = new OidcConfiguration();
+        conf.setClientId(clientId);
+        conf.setSecret(clientSecret.getPlainText());
+
+        // TODO what do we prefer?
+        // conf.setPreferredJwsAlgorithm(JWSAlgorithm.HS256);
+        // set many more as needed...
```

Comment:
not intending for this to happen in this PR but a follow-up that exposes some more advanced options.
e.g. #408